python/tests/operations/unique_metadata.py

```diff
@@ -0,0 +1,705 @@
+        """
+        Tests false of getting unique metadata of gradients on whole TensorMap
+        along properties with incorrect order
+        """
+        # 3 and 4 switched
+        parameter = "parameter"
+        properties, pnames = [0, 4, 3, 5], ["properties"]
+        target_properties = Labels(
+            names=pnames, values=np.array([[p] for p in properties])
+        )
+        actual_properties = fn.unique_metadata(
+            self.tensor1, "properties", pnames, parameter
+        )
+        self.assertFalse(
+            fn._utils._labels_equal(
+                target_properties, actual_properties, exact_order=True
+            )
+        )
+
+
+class TestUniqueMetadataErrors(unittest.TestCase):
+    """
+    Testing the errors raised by :py:func:`unique_metadata` and
+    :py:func:`unique_metadata_block`
+    """
+
+    def setUp(self):
+        self.tensor = equistore.io.load(
+            os.path.join(DATA_ROOT, TEST_FILE),
+            use_numpy=True,
+        )
+        self.block = self.tensor.block(0)
+
+    def test_unique_indices_block_type_errors(self):
+        # TypeError with TM
+        with self.assertRaises(TypeError) as cm:
+            fn.unique_metadata_block(self.tensor, "samples", ["structure"])
+        self.assertEqual(
+            str(cm.exception),
+            "`block` must be an equistore `TensorBlock`",
+        )
+        # TypeError axis as float
+        with self.assertRaises(TypeError) as cm:
+            fn.unique_metadata_block(self.block, 3.14, ["structure"])
+        self.assertEqual(
+            str(cm.exception),
+            "`axis` must be a `str`, either `'samples'` or `'properties'`,"
+            + f" receieved type {type(3.14)}",
+        )
+        # TypeError names as float
+        with self.assertRaises(TypeError) as cm:
+            fn.unique_metadata_block(self.block, "properties", 3.14)
+        self.assertEqual(str(cm.exception), "`names` must be a `list` of `str`")
+        # TypeError gradient_param as float
+        with self.assertRaises(TypeError) as cm:
+            fn.unique_metadata_block(
+                self.block, "properties", ["structure"], gradient_param=3.14
+            )
+        self.assertEqual(
+            str(cm.exception), f"`gradient_param` must be a `str`, not {type(3.14)}"
+        )
+
+    def test_unique_indices_block_value_errors(self):
+        """tests raising of value errors"""
+        # ValueError axis not "samples" or "properties"
+        with self.assertRaises(ValueError) as cm:
+            fn.unique_metadata_block(self.block, "ciao", ["structure"])
+        self.assertEqual(
+            str(cm.exception),
+            "`axis` must be passed as either `'samples'` or `'properties'`,"
+            + " ciao was passed",
+        )
+        # ValueError names not in block - samples
+        with self.assertRaises(ValueError) as cm:
+            fn.unique_metadata_block(self.block, "samples", ["ciao"])
```

Comment:
i like the italian ahahahaha

Reply:
Haha gotta sweeten the deal for my italian reviewers!